src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -1310,19 +1312,21 @@ private static bool SatisfiesConstructorConstraint(TypeSymbol typeArgument)
         }
 
         /// <summary>
-        /// Return true if the class type has a public parameterless constructor.
+        /// Return true if the type has a public parameterless constructor;
+        /// false if the parameterless constructor is not public; and
+        /// null if no parameterless constructor.
         /// </summary>
-        private static bool HasPublicParameterlessConstructor(NamedTypeSymbol type)
+        private static bool? HasPublicParameterlessConstructor(NamedTypeSymbol type)
```

Comment:
nit: For readability, consider passing a parameter instead of returning 3 values, maybe `allowDefaultConstructor`

Reply:
> consider passing a parameter instead of returning 3 values

I think I prefer the current approach. Adding a parameter might make the combinations of parameter and return value just as confusing.

Reply:
I have to agree with Julien. The `true/false/null` return scheme going right now is pretty confusing.